pool/web/view/page/page_admin_assignments.ml

```diff
@@ -26,28 +57,68 @@ module Partials = struct
   ;;
 
   let overview_list
+    redirect
     Pool_context.{ language; csrf; _ }
     experiment_id
     session
     assignments
     =
-    let cancelable = Session.assignments_cancelable session |> CCResult.is_ok in
-    let action assignment =
-      Format.asprintf
-        "/admin/experiments/%s/assignments/%s/cancel"
-        (experiment_id |> Experiment.Id.value)
-        (assignment.Assignment.id |> Pool_common.Id.value)
+    let deletable m = m |> Assignment.is_deletable |> CCResult.is_ok in
+    let cancelable m =
+      Session.assignments_cancelable session |> CCResult.is_ok
+      && Assignment.is_cancellable m |> CCResult.is_ok
+    in
+    let action assignment suffix =
+      assignment_specific_path experiment_id session assignment suffix
       |> Sihl.Web.externalize_path
     in
+    let button_form suffix confirmable control icon assignment =
+      let hidden_redirect_input =
+        let open Pool_common.Message in
+        input_element
+          ~value:(show_assignment_redirect redirect)
+          language
+          `Hidden
+          Field.Redirect
+      in
+      form
+        ~a:
+          [ a_action (action assignment suffix)
+          ; a_method `Post
+          ; a_user_data
+              "confirmable"
+              Pool_common.(Utils.confirmable_to_string language confirmable)
+          ]
+        [ csrf_element csrf ()
+        ; hidden_redirect_input
+        ; submit_element language control ~submit_type:`Error ~has_icon:icon ()
+        ]
+    in
+    let cancel =
+      let open Pool_common in
+      button_form
+        "cancel"
+        I18n.CancelAssignment
+        (Message.Cancel None)
+        `CloseCircle
+    in
+    let mark_as_deleted =
+      let open Pool_common in
+      (* TODO[timhub]: Add hint, only if this assignment has a follow-up
```

Comment:
todo

Reply:
What do you think of this solution? https://github.com/uzh/pool/pull/110/commits/0ba62c704de30bb4f3e30374ba22bbab636eecba

Reply:
@mabiede ping

Reply:
Nice, a small update that makes things easier.